tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/cluster/crossprotocol/ProtocolsMessageLoadBalancingTest.java

```diff
@@ -0,0 +1,279 @@
+
+      pauseClusteringBridges(servers[0]);
+
+
+      // sending Messages.. they should be load balanced
+      {
+         ConnectionFactory cf =  getJmsConnectionFactory(0);
+         Connection cn = cf.createConnection();
+         Session sn = cn.createSession(false, Session.AUTO_ACKNOWLEDGE);
+         MessageProducer pd = sn.createProducer(sn.createQueue(queueName.toString()));
+
+         for (int i = 0; i < NUMBER_OF_MESSAGES; i++) {
+            pd.send(sn.createTextMessage("hello " + i));
+         }
+
+         cn.close();
+      }
+
+      receiveMessages(connection[0], consumer[0], NUMBER_OF_MESSAGES / 2);
+      connection[1].start();
+      Assert.assertNull(consumer[1].receiveNoWait());
+      connection[1].stop();
+
+      servers[0].stop();
+      clearServer(0);
+
+      setupServer(0, isFileStorage(), isNetty());
+      servers[0].addProtocolManagerFactory(new ProtonProtocolManagerFactory());
+
+      setupClusterConnection("cluster0", "queues", MessageLoadBalancingType.STRICT, 1, isNetty(), 0, 1);
+
+      servers[0].start();
+
+      receiveMessages(connection[1], consumer[1], NUMBER_OF_MESSAGES / 2);
+      for (int node = 0; node < NUMBER_OF_SERVERS; node++) {
+         connection[node].close();
+      }
+
+   }
+
+   @Test
+   public void testExpireRedistributed() throws Exception {
+
+      startServers(MessageLoadBalancingType.ON_DEMAND);
+
+      ConnectionFactory factory = getJmsConnectionFactory(1);
+
+
+      waitForBindings(0, "queues.0", 1, 0, true);
+      waitForBindings(1, "queues.0", 1, 0, true);
+
+      waitForBindings(0, "queues.0", 1, 0, false);
+      waitForBindings(1, "queues.0", 1, 0, false);
+
+
+      // sending Messages..
+      {
+         ConnectionFactory cf =  getJmsConnectionFactory(0);
+         Connection cn = cf.createConnection();
+         Session sn = cn.createSession(false, Session.AUTO_ACKNOWLEDGE);
+         MessageProducer pd = sn.createProducer(sn.createQueue(queueName.toString()));
+         pd.setTimeToLive(200);
+
+         for (int i = 0; i < NUMBER_OF_MESSAGES; i++) {
+            pd.send(sn.createTextMessage("hello " + i));
+         }
+
+         cn.close();
+      }
+
+      // time to let stuff expire
+      Thread.sleep(200);
+
+
+      Connection connection = factory.createConnection();
+      Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+      MessageConsumer consumer = session.createConsumer(session.createQueue("queues.expiry"));
+
+      receiveMessages(connection, consumer, NUMBER_OF_MESSAGES);
+      connection.close();
+   }
+
+   private void receiveMessages(Connection connection,
+                                MessageConsumer messageConsumer,
+                                int messageCount) throws JMSException {
+      connection.start();
+
+      for (int i = 0; i < messageCount; i++) {
+         Message msg = messageConsumer.receive(5000);
+      }
+
+      Assert.assertNull(messageConsumer.receiveNoWait());
```

Comment:
I think the tests would be clearer to read if this assert was in the test method, or receiveMessages was renamed to convey it checks there are no more.